edb/schema/pointers.py

```diff
@@ -1484,10 +1532,72 @@ def _alter_begin(
         if orig_target == new_target:
             return schema
 
-        schema = self._propagate_if_expr_refs(
-            schema, context, action=f'alter the type of {vn}')
-
         if not context.canonical:
+            assert orig_target is not None
+            assert new_target is not None
+            pop = self.get_parent_op(context)
+
+            if (
```

Comment:
What is supposed to happen if a USING expression is specified when the value can be casted? It looks like it is silently ignored, which seems bad

Reply:
Yikes. This is unintentional. Cast expr should always be evaluated.